pkg/client/delegation/redelegation.go

```diff
@@ -27,22 +38,54 @@ func (d redelegation) GetGatewayAddress() string {
 }
 
 // newRedelegationEventFactoryFn is a factory function that returns a
-// function that attempts to deserialise the given bytes into a redelegation
+// function that attempts to deserialize the given bytes into a redelegation
 // struct. If the delegate struct has an empty app address then an
 // ErrUnmarshalRedelegation error is returned. Otherwise if deserialisation
 // fails then the error is returned.
 func newRedelegationEventFactoryFn() events.NewEventsFn[client.Redelegation] {
-	return func(redelegationEventBz []byte) (client.Redelegation, error) {
-		redelegationEvent := new(redelegation)
-		if err := json.Unmarshal(redelegationEventBz, redelegationEvent); err != nil {
+	return func(eventBz []byte) (client.Redelegation, error) {
+		txEvent := new(TxEvent)
+		// Try to deserialize the provided bytes into a TxEvent.
+		if err := json.Unmarshal(eventBz, txEvent); err != nil {
 			return nil, err
 		}
-
-		if redelegationEvent.AppAddress == "" || redelegationEvent.GatewayAddress == "" {
-			return nil, events.ErrEventsUnmarshalEvent.
-				Wrapf("with redelegation: %s", string(redelegationEventBz))
+		// Check if the TxEvent has empty transaction bytes, which indicates
+		// the message might not be a valid transaction event.
+		if bytes.Equal(txEvent.Tx, []byte{}) {
+			return nil, events.ErrEventsUnmarshalEvent.Wrapf("%s", string(eventBz))
```

Comment:
NIT: I don't think that a `string(eventBz)` is really useful in logs or return values. I'd provide more semantic data unless you actually found this to be useful.